File: posicao.cpp
/*
Projeto: Jogo de Xadrez
3º Semestre de Ciências da Computação - UFSCar Sorocaba
Programação orientada a objetos
Profª Katti Faceli
Integrantes: Giovanni Alvarenga, Fábio Yamaya, João Victor Vilar
Implementação da classe "posicao"
Descrição:Implementa as funções da classe "posicao" contidas no arquivo .h
*/
#include "posicao.h"

using namespace std;

//Construtor da classe
Posicao::Posicao(char _linha, int _coluna, int _cor) : cor(_cor){
    linha = _linha;
    coluna = _coluna;
    peca = NULL;
}

//Destrutor da classe
Posicao::~Posicao(){

}

//Modifica peca devido ao escopo privado
void Posicao::setPeca(Peca* p){
    peca = p;
}

//Retorna peca devido ao escopo privado
Peca* Posicao::getPeca(){
    return peca;
}

//Retorna a cor devido ao escopo privado
int Posicao::getCor(){
    return cor;
}

//Retorna um valor booleano para saber se uma posição está ocupada ou não
bool Posicao::estaOcupado(){
    if(peca != NULL){
        return true;
    } else {
        return false;
    }
}

File: Posicao.java
public class Posicao {
    private final int cor;
    private int linha; /*
                        * (sera tratado como int para facilitar a implementação, porem os movimentos
                        * das peças serão feitos por char (é simples transforamar de char para int,
                        * assim como foi feito no método que imprime o tabuleiro), neste caso é só
                        * subtrair 97 :)
                        */
    private int coluna;
    private boolean ocupado;
    private Peca peca;

    public Posicao(int _linha, int _coluna, int _cor) {
        this.cor = _cor;
        this.linha = _linha;
        this.coluna = _coluna;
    }

    public int getCor() {
        return this.cor;
    }

    public Peca getPeca(){
        return this.peca;
    }

    public boolean getOcupado() {
        return this.ocupado;
    }

    public void setOcupado() {
        this.ocupado = true;
    }

    public void setLivre() {
        this.ocupado = false;
    }

   //Modifica peca devido ao escopo privado
    public void setPeca(Peca p){
        peca = p;
    }
}
